.github/workflows/ci-verification.yml

```diff
@@ -116,7 +117,7 @@ jobs:
 
       - run: ./tlc.py mc consensus/MCabs.tla
       - run: ./tlc.py --trace-name 1C2N mc --term-count 2 --request-count 2 --raft-configs 1C2N consensus/MCccfraft.tla
-      - run: ./tlc.py --trace-name 1C3N mc --term-count 2 --request-count 2 --raft-configs 1C3N consensus/MCccfraft.tla
+      - run: ./tlc.py --trace-name 1C3N mc --term-count 0 --request-count 3 --raft-configs 1C3N consensus/MCccfraft.tla
```

Comment:
what made it suddenly so much slower? can we cut that instead?

Reply:
It's difficult to tell, it got very fast after https://github.com/microsoft/CCF/pull/6511#discussion_r1782340353, unintentionally. Aside from the config change in that PR, and the wrapper changes in #6513, the only significant change we have done recently is refinement of the abstract spec.

This has definitely made the reconfiguration checks much slower, as document in https://github.com/microsoft/CCF/pull/6509#issuecomment-2387869033, I doubt it's single-handedly responsible for the increase here.

My proposal is:

1. find a level that's workable with the current config (15 minutes tops)
2. test with and without refinement

I am not doing 2hrs+ runs to get to the bottom of this :)

Reply:
I'm not doing 2hr+ run either. How able relocating with term-count 1 to long-verification?

Reply:
I don't want to move stuff over to the long job if we don't know how long it takes. What's already there has a tendency to become much slower on PRs already without us noticing, because it doesn't gate the PRs of course.

Reply:
Did a run of:

```
$ ./tlc.py --trace-name 1C3N mc --term-count 1 --request-count 1 --raft-configs 1C3N consensus/MCccfraft.tla --disable-check-quorum
```

and that is not showing signs of terminating in a practical amount of time:

```
Finished computing initial states: 3 states generated, with 1 of them distinct at 2024-10-07 15:58:36.
Progress(12) at 2024-10-07 15:58:39: 30,499 states generated (30,499 s/min), 9,506 distinct states found (9,506 ds/min), 4,901 states left on queue.
Progress(18) at 2024-10-07 15:59:39: 2,181,569 states generated (2,151,070 s/min), 617,873 distinct states found (608,367 ds/min), 346,123 states left on queue.
Progress(19) at 2024-10-07 16:00:39: 4,354,296 states generated (2,172,727 s/min), 1,211,762 distinct states found (593,889 ds/min), 680,791 states left on queue.
Progress(19) at 2024-10-07 16:01:39: 6,552,748 states generated (2,198,452 s/min), 1,797,566 distinct states found (585,804 ds/min), 1,017,019 states left on queue.
Progress(19) at 2024-10-07 16:02:39: 8,715,079 states generated (2,162,331 s/min), 2,357,474 distinct states found (559,908 ds/min), 1,319,938 states left on queue.
Progress(20) at 2024-10-07 16:03:39: 10,843,863 states generated (2,128,784 s/min), 2,930,425 distinct states found (572,951 ds/min), 1,637,594 states left on queue.
Progress(20) at 2024-10-07 16:04:39: 13,017,919 states generated (2,174,056 s/min), 3,485,983 distinct states found (555,558 ds/min), 1,942,329 states left on queue.
Progress(20) at 2024-10-07 16:05:39: 15,211,885 states generated (2,193,966 s/min), 4,025,951 distinct states found (539,968 ds/min), 2,245,291 states left on queue.
Progress(20) at 2024-10-07 16:06:39: 17,329,118 states generated (2,117,233 s/min), 4,565,008 distinct states found (539,057 ds/min), 2,525,599 states left on queue.
Progress(20) at 2024-10-07 16:07:39: 19,441,359 states generated (2,112,241 s/min), 5,078,066 distinct states found (513,058 ds/min), 2,793,634 states left on queue.
Progress(20) at 2024-10-07 16:08:39: 21,514,517 states generated (2,073,158 s/min), 5,609,653 distinct states found (531,587 ds/min), 3,073,375 states left on queue.
Progress(21) at 2024-10-07 16:09:39: 23,589,086 states generated (2,074,569 s/min), 6,132,063 distinct states found (522,410 ds/min), 3,350,651 states left on queue.
Progress(21) at 2024-10-07 16:10:39: 25,766,479 states generated (2,177,393 s/min), 6,683,257 distinct states found (551,194 ds/min), 3,650,665 states left on queue.
Progress(21) at 2024-10-07 16:11:39: 27,975,357 states generated (2,208,878 s/min), 7,217,003 distinct states found (533,746 ds/min), 3,936,283 states left on queue.
Progress(21) at 2024-10-07 16:12:39: 30,185,423 states generated (2,210,066 s/min), 7,766,557 distinct states found (549,554 ds/min), 4,226,111 states left on queue.
Progress(21) at 2024-10-07 16:13:39: 32,417,193 states generated (2,231,770 s/min), 8,301,938 distinct states found (535,381 ds/min), 4,516,835 states left on queue.
Progress(21) at 2024-10-07 16:14:39: 34,652,394 states generated (2,235,201 s/min), 8,805,080 distinct states found (503,142 ds/min), 4,782,455 states left on queue.
Progress(21) at 2024-10-07 16:15:39: 36,804,114 states generated (2,151,720 s/min), 9,336,156 distinct states found (531,076 ds/min), 5,055,879 states left on queue.
Progress(21) at 2024-10-07 16:16:39: 38,941,101 states generated (2,136,987 s/min), 9,857,079 distinct states found (520,923 ds/min), 5,306,217 states left on queue.
Progress(21) at 2024-10-07 16:17:39: 41,100,611 states generated (2,159,510 s/min), 10,361,344 distinct states found (504,265 ds/min), 5,559,628 states left on queue.
Progress(21) at 2024-10-07 16:18:39: 43,165,057 states generated (2,064,446 s/min), 10,862,363 distinct states found (501,019 ds/min), 5,816,796 states left on queue.
Progress(21) at 2024-10-07 16:19:39: 45,265,180 states generated (2,100,123 s/min), 11,368,679 distinct states found (506,316 ds/min), 6,081,627 states left on queue.
Progress(21) at 2024-10-07 16:20:39: 47,364,110 states generated (2,098,930 s/min), 11,873,632 distinct states found (504,953 ds/min), 6,324,049 states left on queue.
```

Reply:
I noticed that we did not have disable check quorum, so I ran

```
./tlc.py --trace-name 1C3N mc --disable-check-quorum --term-count 1 --request-count 1 --raft-configs 1C3N consensus/MCccfraft.tla
```

alas:

```
Progress(13) at 2024-10-07 16:21:29: 48,575 states generated (48,575 s/min), 14,608 distinct states found (14,608 ds/min), 7,350 states left on queue.
Progress(18) at 2024-10-07 16:22:29: 2,485,611 states generated (2,437,036 s/min), 703,478 distinct states found (688,870 ds/min), 392,735 states left on queue.
Progress(19) at 2024-10-07 16:23:29: 4,974,280 states generated (2,488,669 s/min), 1,378,931 distinct states found (675,453 ds/min), 775,160 states left on queue.
Progress(19) at 2024-10-07 16:24:29: 7,491,724 states generated (2,517,444 s/min), 2,046,296 distinct states found (667,365 ds/min), 1,148,980 states left on queue.
Progress(20) at 2024-10-07 16:25:29: 9,990,153 states generated (2,498,429 s/min), 2,697,360 distinct states found (651,064 ds/min), 1,512,586 states left on queue.
Progress(20) at 2024-10-07 16:26:29: 12,461,482 states generated (2,471,329 s/min), 3,343,114 distinct states found (645,754 ds/min), 1,848,279 states left on queue.
Progress(20) at 2024-10-07 16:27:29: 14,975,666 states generated (2,514,184 s/min), 3,975,135 distinct states found (632,021 ds/min), 2,190,025 states left on queue.
Progress(20) at 2024-10-07 16:28:29: 17,542,319 states generated (2,566,653 s/min), 4,619,348 distinct states found (644,213 ds/min), 2,550,124 states left on queue.
Progress(21) at 2024-10-07 16:29:29: 20,048,771 states generated (2,506,452 s/min), 5,232,347 distinct states found (612,999 ds/min), 2,868,352 states left on queue.
Progress(21) at 2024-10-07 16:30:29: 22,560,003 states generated (2,511,232 s/min), 5,876,300 distinct states found (643,953 ds/min), 3,232,098 states left on queue.
Progress(21) at 2024-10-07 16:31:29: 25,023,817 states generated (2,463,814 s/min), 6,518,528 distinct states found (642,228 ds/min), 3,570,854 states left on queue.
Progress(21) at 2024-10-07 16:32:29: 27,506,196 states generated (2,482,379 s/min), 7,125,180 distinct states found (606,652 ds/min), 3,867,237 states left on queue.
Progress(21) at 2024-10-07 16:33:29: 29,926,949 states generated (2,420,753 s/min), 7,730,783 distinct states found (605,603 ds/min), 4,181,736 states left on queue.
Progress(21) at 2024-10-07 16:34:29: 32,410,676 states generated (2,483,727 s/min), 8,313,189 distinct states found (582,406 ds/min), 4,473,191 states left on queue.
Progress(21) at 2024-10-07 16:35:29: 34,949,994 states generated (2,539,318 s/min), 8,907,351 distinct states found (594,162 ds/min), 4,785,707 states left on queue.
Progress(21) at 2024-10-07 16:36:29: 37,532,679 states generated (2,582,685 s/min), 9,519,414 distinct states found (612,063 ds/min), 5,105,041 states left on queue.
Progress(21) at 2024-10-07 16:37:29: 40,076,444 states generated (2,543,765 s/min), 10,124,037 distinct states found (604,623 ds/min), 5,425,704 states left on queue.
Progress(22) at 2024-10-07 16:38:29: 42,584,693 states generated (2,508,249 s/min), 10,715,356 distinct states found (591,319 ds/min), 5,728,721 states left on queue.
Progress(22) at 2024-10-07 16:39:29: 45,073,294 states generated (2,488,601 s/min), 11,323,254 distinct states found (607,898 ds/min), 6,039,907 states left on queue.
Progress(22) at 2024-10-07 16:40:29: 47,590,025 states generated (2,516,731 s/min), 11,939,602 distinct states found (616,348 ds/min), 6,381,284 states left on queue.
Progress(22) at 2024-10-07 16:41:29: 50,092,469 states generated (2,502,444 s/min), 12,552,310 distinct states found (612,708 ds/min), 6,709,491 states left on queue.
Progress(22) at 2024-10-07 16:42:29: 52,564,098 states generated (2,471,629 s/min), 13,181,518 distinct states found (629,208 ds/min), 7,034,326 states left on queue.
Progress(22) at 2024-10-07 16:43:29: 55,018,985 states generated (2,454,887 s/min), 13,803,539 distinct states found (622,021 ds/min), 7,358,512 states left on queue.
Progress(22) at 2024-10-07 16:44:29: 57,529,645 states generated (2,510,660 s/min), 14,377,824 distinct states found (574,285 ds/min), 7,628,531 states left on queue.
Progress(22) at 2024-10-07 16:45:29: 59,994,261 states generated (2,464,616 s/min), 14,960,246 distinct states found (582,422 ds/min), 7,890,918 states left on queue.
Progress(22) at 2024-10-07 16:46:29: 62,417,677 states generated (2,423,416 s/min), 15,535,113 distinct states found (574,867 ds/min), 8,153,813 states left on queue.
Progress(22) at 2024-10-07 16:47:29: 64,921,718 states generated (2,504,041 s/min), 16,125,859 distinct states found (590,746 ds/min), 8,457,972 states left on queue.
Progress(22) at 2024-10-07 16:48:29: 67,426,653 states generated (2,504,935 s/min), 16,704,882 distinct states found (579,023 ds/min), 8,748,055 states left on queue.
Progress(22) at 2024-10-07 16:49:29: 69,988,510 states generated (2,561,857 s/min), 17,270,696 distinct states found (565,814 ds/min), 9,007,120 states left on queue.
```